packages/widgetbook/test/src/knobs/knobs_test.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:provider/provider.dart';
+import 'package:widgetbook/src/knobs/knobs.dart';
+import 'package:widgetbook/src/repositories/selected_story_repository.dart';
+import 'package:widgetbook/widgetbook.dart';
+
+import '../../helper/widget_test_helper.dart';
+
+void main() {
+  testWidgets(
+    'Bool knob added',
+    (WidgetTester tester) async {
+      final selectedStoryRepository = SelectedStoryRepository();
+      final knobsNotifier = KnobsNotifier(selectedStoryRepository);
+      final useCase = WidgetbookUseCase(
+        name: 'use case',
+        builder: (context) {
+          return Column(
+            children: [
+              Text(context.knobs.boolean(
+                label: 'label',
+                initial: true,
+              )
+                  ? 'Hi'
+                  : 'Bye'),
+              ...knobsNotifier.all().map((e) => e.build())
+            ],
+          );
+        },
+      );
+      selectedStoryRepository.setItem(useCase);
+      await tester.pumpWidgetWithMaterialApp(ChangeNotifierProvider(
+        create: (context) => knobsNotifier,
+        child: Builder(builder: useCase.builder),
+      ));
+      expect(knobsNotifier.all().length, equals(1));
```

Comment:
I'd also test for the state like this, because it is more precise:

```
expect(
    knobsNotifier.all(),
    equals(
      [
        BoolKnob(label: 'label', value: true),
      ],
    ),
  ); 
```

For this the equality operators have to be overridden - `freezed` might help with that.